app/Services/PlanetService.php

```diff
@@ -990,6 +991,27 @@ public function addResources(Resources $resources, bool $save_planet = true): vo
         }
     }
 
+    /**
+     * @param string $resourceName
+     * @param int|float $amount
+     * @param bool $save_planet
+     * @return void
+     * @throws Exception
+     */
+    public function addResource(string $resourceName, int|float $amount, bool $save_planet = true): void
+    {
+
+        if (in_array($resourceName, array_column(ResourceType::cases(), "value")) && isset($this->planet->{$resourceName})) {
+            $this->planet->{$resourceName} += $amount;
+            if ($save_planet) {
+                $this->save();
+            }
+        } else {
+            throw new Exception('Invalid Resource');
+        }
+
+    }
+
```

Comment:
Nice idea to add a separate method for the addResource(), will come in handy in other places as well  that are only adding or subtracting a single resource type.

I do would like to add a suggestion: could you change the `$resourceName` parameter in this method to be a `ResourceType` enum case instead of a string. This so we have static type safety for all places that call this method. I did see you added a test for the invalid resource name detection behaviour which is good too. But if we have static types then PhpStan will be able to detect any incorrect callers automatically during testing.

After this change you will probably have to change the DeveloperShortcutsController.php file too and do the string to enum mapping there before calling the `addResource` method. All in all I think this would make it more robust.